pkg/vminetworkscontroller/vmi_controller.go

```diff
@@ -83,44 +83,31 @@ func (r *VirtualMachineInstanceReconciler) Reconcile(
 	}
 
 	ownerInfo := ownerReferenceFor(vmi, vm)
-	for logicalNetworkName, netConfigName := range vmiNetworks {
-		claimKey := fmt.Sprintf("%s.%s", vmi.Name, logicalNetworkName)
-		ipamClaim := &ipamclaimsapi.IPAMClaim{
-			ObjectMeta: controllerruntime.ObjectMeta{
-				Name:            claimKey,
-				Namespace:       vmi.Namespace,
-				OwnerReferences: []metav1.OwnerReference{ownerInfo},
-				Finalizers:      []string{claims.KubevirtVMFinalizer},
-				Labels:          claims.OwnedByVMLabel(vmi.Name),
-			},
-			Spec: ipamclaimsapi.IPAMClaimSpec{
-				Network: netConfigName,
-			},
-		}
 
-		if err := r.Client.Create(ctx, ipamClaim, &client.CreateOptions{}); err != nil {
-			if apierrors.IsAlreadyExists(err) {
-				claimKey := apitypes.NamespacedName{
-					Namespace: vmi.Namespace,
-					Name:      claimKey,
-				}
+	// UDN code block
+	primaryUDN, err := r.vmiPrimaryUDN(ctx, vmi)
+	if err != nil {
+		return controllerruntime.Result{}, err
+	}
+	if primaryUDN != nil {
+		claimKey := fmt.Sprintf("%s.%s-primary-udn", vmi.Name, primaryUDN.Name)
```

Comment:
We should normalize the key to follow k8s naming conventions, I have see errors like the following when the network name at the NAD has unexpected chars.

```
 Retry add failed for *v1.Pod kv-live-migration-8805/virt-launcher-worker1-987pq, will try again later: failed to update pod kv-live-migration-8805/virt-launcher-worker1-987pq: error retrieving IPAMClaim for pod kv-live-migration-8805/virt-launcher-worker1-987pq: failed to get IPAMClaim "worker1.jv45k_net1-primary-udn": ipamclaim.k8s.cni.cncf.io "worker1.jv45k_net1-primary-udn" not found
```